### ISSUE TITLE
Thanks ♥

### ISSUE THREAD

**Issue by ekruijf** [tjarb/duco.modbus#1]
Dear Tjarb,

Thank you immensely for developing the Duco app for Homey. Your contribution fills a long-standing gap within the Homey community, offering a much-needed solution for integrating Duco ventilation systems. I personally own a DucoBox Energy Premium and have already installed a communication print in anticipation of enhancing its connectivity. Although I'm not a developer, I'm eager to assist in any way I can to support the inclusion of the DucoBox Energy Premium within your app. Your work is greatly appreciated, and I look forward to any possibility of contributing to this project.

Best regards.

**Comment by tjarb** [comment]
Hi,

I'm currently a bit limited in time, but this week we have pinpointed the
difference between modbus RTU en the Modbus TCP Connectivity PCB. So it
will be a matter of time for me to implement / extend this.

Please be patience.

Best regards

Op za 2 mrt 2024 15:06 schreef ekruijf ***@***.***>:

**Comment by ekruijf** [comment]
Hi Tjarb,

I will be! Thank you!

**Comment by erwindek** [comment]
Hi Tjard,

How are you doing? In our previous contact we talked about the expansion of the Duco app. You then indicated that you had already made progress with this. Do you still need help to complete this further? I also noticed that the app still has experimental status. Has this app already been offered for approval to Homey?

**Comment by tjarb** [comment]
Hello Erwin,

thank you for your message. I'm very limited in time so made not much
progress.
The experimental is because it's not (yet) compatible with Duco's latest
version of modbus TCP server and the implementation is Read Only for now.
I need a day to add devices based on the "new" register structures. And an
additional discovery method as address spacing is different. Then I can
release a new version for testing.

Best regards,

Op zo 12 mei 2024 08:37 schreef erwindek ***@***.***>:

**Comment by tjarb** [comment]
Hi Erwin,

I've done some work this morning. Version 3.0 is available now. I added
discovery for TCP address spacing. You may try it.

If you find devices (staring with the master box) I can move forward
implementing nodes and their endpoints.

Best regards,

Op zo 12 mei 2024 10:50 schreef Tjarco Boerkoel ***@***.***>:

**Comment by erwindek** [comment]
![WhatsApp Image 2024-05-30 at 10 29 32](https://github.com/tjarb/duco.modbus/assets/52920445/8e88b328-fc70-4b96-9619-f6c2431c16e5)
Hi Tjard,

How good to hear that you have found time to add some developments! Thanks for that.

I just tried to add the Master unit. After entering the IP address and leaving the default values ​​unchanged, I get an error. "Could not open device on ..."

I can access the unit via the browser via that IP.

Can I try something different?

Best regards,
Erwin

**Comment by lvdp1234** [comment]
Does the app still work? 

I have trouble with getting up and running. I have: 

Device: DucoBox energy Comfort 325 L
Connectivity Board: Duco Connectivity Board (0000-4810)
Homey-app: DUCO-app installed
2x bediendingsschakelaar Battery
2x CO2 sensor

I connected it via UTP cable. Everything works fine. Until the app starts Discovery mode. Then it sees 5 devices: “Unknown devices”. When I add them all, I get the message: The ‘id’ argument must be of type string. Received null

Does anyone know a solution? I've tried so many things already.
